### ISSUE TITLE
Failed to deploy virtual kubelet when the AKS cluster has MSI enabled

### ISSUE THREAD

**Issue by gossion** [virtual-kubelet/azure-aci#123]
Follow this guidance https://github.com/virtual-kubelet/azure-aci#deployment-of-the-aci-provider-in-your-cluster to deploy virtual kubelet on AKS which has MSI enabled, we will get the error like 
```
default       virtual-kubelet3azure-virtual-kubelet-aci-for-aks-b4d85898fsjmd   0/1     CrashLoopBackOff   1          6s



$ k logs virtual-kubelet3azure-virtual-kubelet-aci-for-aks-b4d85898fsjmd
....
time="2021-02-02T07:55:26Z" level=fatal msg="error initializing provider azure: Neither AZURE_CLIENT_ID or VIRTUALNODE_USER_IDENTITY_CLIENTID is being set"
```

install commands:
```
helm install "$RELEASE_NAME" "$CHART_URL" \
  --set provider=azure \
  --set providers.azure.targetAKS=true \
  --set providers.azure.masterUri=$MASTER_URI \
  --set nodeName=$NODE_NAME
```

The problem does not exist when the AKS cluster uses SP.

**Comment by feiskyer** [comment]
@gossion this should be fixed by #125?

**Comment by gossion** [comment]
yes. closing it.
